### PR TITLE
[PM-8593] CLI - Logout needs to reset active account

### PR DESCRIPTION
**Pull request by ttalty**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8593

## 📔 Objective

When the user was logging out the account service was clearing any active user info but not setting the active user to null. When changing any server config setting. The logged out user's settings were being applied. When attempting to change the settings while logged out. The previously logged in user's settings would be updated since the active account was still set to that account. 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/cli/src/service-container.ts [hunk 1]
```diff
@@ -754,6 +751,9 @@ export class ServiceContainer {
     await this.stateService.clean();
     await this.accountService.clean(userId);
     process.env.BW_SESSION = null;
+    this.authService.logOut(async () => {
+      await this.accountService.switchAccount(null);
```
Comment (ttalty) [inline apps/cli/src/service-container.ts:755]: One possible change here. Should the account switch call be made within the auth service logout?

**Comment by eliykat** [comment]
Hey @ttalty, I assume you requested my review because I refactored this file recently, however I think Platform are more knowledgeable about account switching in particular. I'm going to request a review from them instead, but if you did want me to look at it specifically, please let me know.

**Review by justindbaur — COMMENTED** [review]
Thanks for finding this fix, I think since the callback given to AuthService isn't awaited there would be a chance at it not finishing before the process exits. 

I think the `AuthService.logOut` call can be left where it is and still empty but the `switchAccount(null)` call can be done right after `AccountService.clean`.

**Review by MGibson1 — APPROVED** [review]
It looks like it is a good idea to add this to the web `app.component` logout as well.

**Pull request merged**